index.js

```diff
@@ -82,18 +94,30 @@ class Pipe2Jpeg extends Transform {
           this._size += sliced.length;
           this._jpeg = Buffer.concat(this._chunks, this._size);
           this._chunks = [];
+          this._lastSize = this._size;
           this._size = 0;
           this._sendJpeg();
           if (pos === chunkLength) {
             break;
           }
         }
       } else {
+        if (this._lastByte === _SOI[0] && chunk[0] === _SOI[1]) {
```

Comment:
We could check that pos > 0 and we could skip this block. Also, we would not need:

`pos = Math.max(pos - 1, 0);`

Reply:
good thought!